raiden/network/transport/matrix/layer.py

```diff
@@ -0,0 +1,108 @@
+import os
+import sys
+from typing import Dict, Any, List
+
+import click
+
+from raiden.constants import PATH_FINDING_BROADCASTING_ROOM, MONITORING_BROADCASTING_ROOM
+from raiden.exceptions import RaidenError
+from raiden.network.transport import MatrixTransport
+from raiden.network.transport.matrix import MatrixLightClientTransport
+from raiden.network.transport.matrix.utils import get_available_servers_from_config, server_is_available
+from raiden.settings import DEFAULT_MATRIX_KNOWN_SERVERS
+from raiden.storage import sqlite, serialize
+from raiden.utils import Address
+from raiden.utils.cli import get_matrix_servers
+from transport.layer import Layer as TransportLayer
+from transport.node import Node as TransportNode
+
+
+class MatrixLayer(TransportLayer):
+
+    def __init__(self, config: Dict[str, Any]):
```

Comment:
this should do everything `_setup_matrix` did, but with a constructor structure now